pkgs/by-name/co/cosmic-tasks/package.nix

```diff
@@ -0,0 +1,99 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, git
+, pkg-config
+, wrapGAppsHook
+, atk
+, cairo
+, gdk-pixbuf
+, glib
+, gtk3
+, libsecret
+, libxkbcommon
+, openssl
+, pango
+, sqlite
+, vulkan-loader
+, stdenv
+, darwin
+, wayland
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cosmic-tasks";
+  version = "unstable-2024-04-11";
+
+  src = fetchFromGitHub {
+    owner = "edfloreshz";
+    repo = "cosmic-tasks";
+    rev = "3b3c8ef609cb27ddcc3510f0d5ca28afb286587f";
+    hash = "sha256-qri9nRQIukFG/ZfiYO4lWHLBP6gFv0GgqLs2sbSwzR8=";
+    #
+    leaveDotGit = true;
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "accesskit-0.12.2" = "sha256-ksaYMGT/oug7isQY8/1WD97XDUsX2ShBdabUzxWffYw=";
+      "atomicwrites-0.4.2" = "sha256-QZSuGPrJXh+svMeFWqAXoqZQxLq/WfIiamqvjJNVhxA=";
+      "clipboard_macos-0.1.0" = "sha256-8lGz40jPdCmsRSoEgS4YWl+TkrYII1906ZVDokM3NMU=";
+      "cosmic-config-0.1.0" = "sha256-PiYcdU8OfeO7OPFp3Jhoev4OIGNSQMGcuPsWPy9usnw=";
+      "cosmic-text-0.11.2" = "sha256-gUIQFHPaFTmtUfgpVvsGTnw2UKIBx9gl0K67KPuynWs=";
+      "d3d12-0.19.0" = "sha256-usrxQXWLGJDjmIdw1LBXtBvX+CchZDvE8fHC0LjvhD4=";
+      "done_core-0.1.0" = "sha256-9PK+ddi6j1e6NQLflOkPgMRCfCNz1QAG8MPCtDznsSo=";
+      "glyphon-0.5.0" = "sha256-j1HrbEpUBqazWqNfJhpyjWuxYAxkvbXzRKeSouUoPWg=";
+      "smithay-client-toolkit-0.18.0" = "sha256-/7twYMt5/LpzxLXAQKTGNnWcfspUkkZsN5hJu7KaANc=";
+      "smithay-clipboard-0.8.0" = "sha256-LDd56TJ175qsj2/EV/dbBRV9HMU7RzgrG5JP7H2PmhE=";
+      "softbuffer-0.4.1" = "sha256-a0bUFz6O8CWRweNt/OxTvflnPYwO5nm6vsyc/WcXyNg=";
+      "taffy-0.3.11" = "sha256-SCx9GEIJjWdoNVyq+RZAGn0N71qraKZxf9ZWhvyzLaI=";
+      "winit-0.29.10" = "sha256-ScTII2AzK3SC8MVeASZ9jhVWsEaGrSQ2BnApTxgfxK4=";
+    };
+  };
+
+  nativeBuildInputs = [
+    git
+    pkg-config
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    atk
+    cairo
+    gdk-pixbuf
+    glib
+    gtk3
+    libsecret
+    libxkbcommon
+    openssl
+    pango
+    sqlite
+    vulkan-loader
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.AppKit
+    darwin.apple_sdk.frameworks.CoreFoundation
+    darwin.apple_sdk.frameworks.CoreGraphics
+    darwin.apple_sdk.frameworks.CoreServices
+    darwin.apple_sdk.frameworks.Foundation
+    darwin.apple_sdk.frameworks.Metal
+    darwin.apple_sdk.frameworks.QuartzCore
+    darwin.apple_sdk.frameworks.Security
+  ] ++ lib.optionals stdenv.isLinux [
+    wayland
+  ];
+
+  postFixup = ''
```

Comment:
I marked the packages as linux-only (`meta.platforms = platforms.linux`).
Do you think it can be used on Darwin ?

Reply:
I don't think so?

Reply:
Just remove the wrap program and this will be good to go.

Reply:
> Just remove the wrap program and this will be good to go.

We do need this patch, right ? (otherwise it does crash at runtime).
Do you suggest changing the `wrapProgram` to something else ?

Reply:
It shouldn't be needed, if it does then it's possibly an upstream bug. It's actually bad practice to wrap your program to add dependencies.

Reply:
Actually, I copied this snippet from another cosmic application derivation.
It looks like most of them need to be patched this way:
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/co/cosmic-edit/package.nix#L84-L101
- https://github.com/NixOS/nixpkgs/blob/61153af0687961878831e00f2de200d9bc2d8224/pkgs/by-name/co/cosmic-files/package.nix#L64-L69
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/co/cosmic-launcher/package.nix#L59-L62
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/co/cosmic-notifications/package.nix#L61-L64
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/co/cosmic-store/package.nix#L75-L90
- https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/co/cosmic-term/package.nix#L95-L108

Reply:
Interesting, well that's fine then.